### PR TITLE
Button to add new property to an Entity List

### PR DESCRIPTION
**Pull request by ktuite**
Closes https://github.com/getodk/central/issues/576

Screenshot 2024-04-01 at 1 38 59 PM
Screenshot 2024-04-01 at 1 39 35 PM

#### Before submitting this PR, please make sure you have:

- [ ] run `npm run test` and `npm run lint` and confirmed all checks still pass OR confirm CircleCI build passes
- [ ] verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

File: src/components/dataset/overview/new-property.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+<template>
+  <modal id="dataset-property-new" :state="state" :hideable="!awaitingResponse" backdrop
+    @hide="$emit('hide')">
+    <template #title>{{ $t('title') }}</template>
+    <template #body>
+      <div class="modal-introduction">
+        <p>{{ $t('introduction[0]') }}</p>
+      </div>
+      <form @submit.prevent="submit">
+        <form-group ref="name.value" v-model.trim="name"
```
Comment (ktuite) [inline src/components/dataset/overview/new-property.vue:21]: Change to be like new dataset PR

File: src/components/dataset/overview/new-property.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+};
+</script>
+
+<i18n lang="json5">
+{
+  "en": {
+    // This is the title at the top of a pop-up.
+    "title": "Add Entity Property",
+    "introduction": [
+      "To add an Entity Property, choose a unique property name below.",
```
Comment (ktuite) [inline src/components/dataset/overview/new-property.vue:90]: In this case, Property should not be capitalized (paralleling 'field')

File: src/components/dataset/overview/dataset-properties.vue [hunk 3]
```diff
@@ -87,13 +88,19 @@ const { publishedFormPath } = useRoutes();
...
+  }
 }
 </style>
 
 <i18n lang="json5">
   {
     "en": {
-      "emptyTable": "The Entities in this Entity List do not have any user-defined properties."
+      "emptyTable": "The Entities in this Entity List do not have any user-defined properties.",
+      // This is shown in a column about no Forms updating the Entity Property in this row
```
Comment (ktuite) [inline src/components/dataset/overview/dataset-properties.vue:102]: Possibly clarify that None refers to Forms (and add period)

File: test/components/dataset/overview.spec.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  describe('new property button', () => {
+    it('allows admins to see new property button', async () => {
+      mockLogin({ role: 'admin' });
+      testData.extendedDatasets.createPast(1, { name: 'trees' });
+      const app = await load('/projects/1/entity-lists/trees');
+      app.find('#dataset-list-new-button').exists().should.be.true();
+    });
+
+    it('does not allow project viewers to see new property button', async () => {
+      mockLogin({ role: 'viewer' });
```
Comment (ktuite) [inline test/components/dataset/overview.spec.js:15]: Consider changing to have a project-specific verb with sitewide role: none.

File: test/data/datasets.js [hunk 5]
```diff
@@ -39,3 +39,10 @@ export const extendedDatasets = dataStore({
   }),
   sort: comparator((dataset1, dataset2) => dataset1.name < dataset2.name)
 });
+
+extendedDatasets.addProperty = (index, property) => {
+  const { properties } = extendedDatasets.get(index);
```
Comment (ktuite) [inline test/data/datasets.js:44]: `const properties = [...extendedDatasets.get(index).properties, normalizeProperty({ name: property })]`

**Review by ktuite — COMMENTED** [review]
interactive review with @matthew-white

**Review by matthew-white — APPROVED** [review]

**Pull request merged**